src/mds/CInode.cc

```diff
@@ -697,48 +697,36 @@ frag_t InodeStoreBase::pick_dirfrag(std::string_view dn)
   return dirfragtree[h];
 }
 
-std::pair<bool, std::vector<CDir*>> CInode::get_dirfrags_under(frag_t fg)
+std::vector<CDir*> CInode::get_dirfrags_under(frag_t fg, bool* all )
 {
-  std::pair<bool, std::vector<CDir*>> result;
-  auto& all = result.first;
-  auto& dirs = result.second;
+  std::vector<CDir*> dirs;
```

Comment:
I don't really see a reason to change the return type. Can you please revert this.

Reply:
Also: `all` should always be true unless we're returning from the end of this method. Might as well set it `true` here.

Reply:
en. 

> I don't really see a reason to change the return type. Can you please revert this.
At the beginning of the period, I expect to set the return value of all as optional. This return value is only used in directory consolidation calculation. In the case, there is a more optimized method to calculate it.  I will revert this.

Reply:
> Also: `all` should always be true unless we're returning from the end of this method. Might as well set it `true` here.

The return value of "all" means whether the partition in memory completely contains the input partition. The original method is to find whether the partition is not completely contained in memory, so firstly set it to true. Now the improvement The method is to calculate whether the current memory contains this partition completely, so the default is incomplete firstly